### DISCUSSION TITLE
Empty lines in snipMate snippets are expanded

### DISCUSSION THREAD

**Discussion by raine** [dcampos/nvim-snippy#82] [General]
If you use snippets such as [these](https://raw.githubusercontent.com/honza/vim-snippets/master/snippets/go.snippets) with snippy, you'll notice that the empty lines after the snippets (before the next snippet) are expanded as well. 

Am I missing something obvious or is this a bug? A simple fix is to manually remove every blank line but the README claims snipMate compatibility so I was wondering about this.

![Kapture 2022-10-11 at 16 05 15](https://user-images.githubusercontent.com/11027/195098731-6353fd8d-1f19-40c8-8fe0-7362bc75a4ca.gif)

^ Notice the extra empty line added when the snippet expands.

Thanks!

**Comment by dcampos** [comment]
There appears to be no mention in its docs or codebase about trailing empty lines, so I had to do some testing to determine this, but it seems like SnipMate does remove a single at the end, while Ultisnips doesn't. So, yes, to be compatible with snipMate, and also support vim-snippets properly, I think Snippy should remove the last line if empty.

Looks like almost all snippet files from vim-snippets avoid trailing empty lines, `go.snippets` being an exception. The blank lines seem to have been introduced by [this commit](https://github.com/honza/vim-snippets/commit/b51c01137259ac7cd59381001279fa36baefbc5f).

**Reply by raine** [reply]
Nice find. I didn't realize go.snippets was an exception in that regard. I think the empty line allows for slightly more readable snippet files. And it sort of makes sense that any line after `snippet ..` that doesn't have a tab as first character is not part of the snippet.
